recipes/dotnet_agent.rb

```diff
@@ -0,0 +1,102 @@
+agent = node['appdynamics']['dotnet_agent']
+controller = node['appdynamics']['controller']
+proxy = node['appdynamics']['http_proxy'] 
+bitness = node['kernel']['os_info']['os_architecture']
+install_directory = agent['install_dir']
+system_directory = node['kernel']['os_info']['windows_directory']
+
+agent_msi = ""
+##################################################################################################################################
+# The below paths can also be used instead of temp direcotry. But there will be msiexec exceptions if there is space in the path #
+# agent_msi_path = "#{Chef::Config[:file_cache_path]"																			 #
+# setup_config = "#{Chef::Config[:file_cache_path]}\\setup.xml"																	 #
+# install_log_file = "#{Chef::Config[:file_cache_path]}\\DotnetAgentInstall.log"												 #
+##################################################################################################################################
+agent_msi_path = "#{system_directory}\\Temp"
+setup_config = "#{agent_msi_path}\\setup.xml"
+install_log_file = "#{agent_msi_path}\\DotnetAgentInstall.log"
+
+
+# Check the bitness of the OS to determine the installer to download and run
+case bitness
+when /64/
+  agent_msi = "#{agent_msi_path}\\dotNetAgentSetup64.msi"
+  source_path = "#{agent['source']}/dotNetAgentSetup64.msi"
+when /32/
+  agent_msi = "#{agent_msi_path}\\dotNetAgentSetup.msi"
+  source_path = "#{agent['source']}/dotNetAgentSetup.msi"
+else
+  agent_msi = "Unsupported OS bitness"
+end
+
+# Download the msi file from source
+remote_file "#{agent_msi}" do
+  source source_path
+  checksum agent['checksum']
+end
+
+# Environment Validation
+# Comment if you're installing 4.1 -- Matt Jensen to confirm with Sanjay
+# Check whether the MSDTC service is up and running 
+# if it is not running, start the service
+powershell_script 'check_MSDTC' do
+  code 'Start-Service MSDTC'
+only_if {'(Get-Service MSDTC).status' != "Running" }
+end
+
+# Comment if you're installing 4.1 -- Matt Jensen to confirm with Sanjay
+# Check whether the WMI service is up and running 
+# if it is not running, start the service
+powershell_script 'check_WMI' do
+  code 'Start-Service Winmgmt'
+only_if {'(Get-Service Winmgmt).status' != "Running" }
+end
+
+# Comment if you're installing 4.1 -- Matt Jensen to confirm with Sanjay
+# Check whether the COM+ service is up and running
+# if it is not running, start the service
+powershell_script 'check_complus' do
+  code 'Start-Service COMSysApp'
+only_if {'(Get-Service COMSysApp).status' != "Running" }
+end
+
+# Check whether IIS 7.0+ is installed
+# Enable IIS Health Monitoring for the Machine snapshots to return IIS App Pool data
+powershell_script 'check_IIS' do
```

Comment:
@thiru-chidambaram  , i'm not sure about the: select versionstring).VersionString.Substring(8)) -ge 7.0' 

what if its IIS7.5 or IIS8? maybe its better to just require this on windows 2008 r2+

i also just realized that the windows cookbook has the powershell for the feature install

@justinredd what do you think is a good way to do this? the requirement is that it should install this feature on windows 2008+ but not on windows 2003(maybe we dont care since windows 2003 is losing support?)
i feel we could check platform_version or something close to that
then do 
windows_feature 'Web-Request-Monitor' do
  action :install
end